thunder/core/proxies.py

```diff
@@ -1354,6 +1361,10 @@ def true_dtype(self):
     def requires_grad(self):
         return self._requires_grad
 
+    @property
+    def grad(self):
+        return None  # FutureTensorProxies never require grad
+
```

Comment:
Let's raise an error for this case.

Reply:
When making a `TensorProxy` or a `FutureTensorProxy` that reflects another `TensorLike`, we copy its metadata and `grad`.
https://github.com/Lightning-AI/lightning-thunder/pull/1070/files#diff-f25d3a604208b167d7f7882a364dd30b8bb9fe395a6cb05b9ed1ffc500ff8e87R1227-R1234
Although `FutureTensorProxy` simply ignores what `_infer_tensor_properties` infers as `grad` ([ignored here](https://github.com/Lightning-AI/lightning-thunder/pull/1070/files#diff-f25d3a604208b167d7f7882a364dd30b8bb9fe395a6cb05b9ed1ffc500ff8e87R1321)), setting `FutureTensorProxy.grad` as `None` keeps this logic easy.

This makes sense to me, because `FutureTensorProxy.requires_grad` is `False` anyway.